### PR TITLE
Chunk Lookup Data

### LINKED ISSUE #3719: Get `n` chunks of Lookup Data
Instead of [encoding lookup data](https://github.com/project-oak/oak/blob/92ffc8d747db9ceae2cdf6f4f1dbb6eea7bfb4d4/oak_functions_launcher/src/lookup.rs#L44) and failing if the size of a chunk of lookup data exceeds, we will return `n` chunks of lookup data which are smaller than `max_chunk size`.

### PR DESCRIPTION
**Pull request by mariaschett**
Fixes #3719 .

### CODE REVIEW

File: oak_functions_launcher/src/lookup.rs [hunk 1]
```diff
@@ -33,9 +33,18 @@ pub async fn update_lookup_data(
...
     let lookup_data = load_lookup_data(lookup_data_path)?;
-    let chunk = encode_lookup_data(lookup_data, max_chunk_size)?;
+    let mut chunks = chunk_up_lookup_data(lookup_data, max_chunk_size);
+
+    if chunks.len() != 1 {
+        return Err(anyhow!(
+            "lookup data size exceeds: {:?} bytes",
+            max_chunk_size
+        ));
+    }
```
Comment (tiziano88) [inline oak_functions_launcher/src/lookup.rs:43]: I thought we were trying to get multiple chunks (the issue even says `n` chunks). Or is this PR not addressing #3719?

File: oak_functions_launcher/src/lookup.rs [hunk 2]
```diff
@@ -130,15 +140,17 @@ fn test_encode_lookup_data_exceed_bound() {
...
 #[test]
-fn test_encode_lookup_data_empty() {
+fn test_chunk_up_lookup_data_empty() {
     let max_chunk_size = ByteUnit::Kibibyte(1);
     let data = hashbrown::HashMap::new();
-    let result = encode_lookup_data(data, max_chunk_size);
-    assert!(result.is_ok());
-    assert_eq!(result.unwrap().items.len(), 0)
+    let chunks = chunk_up_lookup_data(data, max_chunk_size);
+    assert!(chunks.len() == 1);
```
Comment (tiziano88) [inline oak_functions_launcher/src/lookup.rs:154]:
```suggestion
    assert_eq!(chunks.len(), 1);
```

**Review by tiziano88 — APPROVED** [review]

File: oak_functions_launcher/src/lookup.rs (hunk 2 above)
Comment (mariaschett) [reply oak_functions_launcher/src/lookup.rs]: Done.

File: oak_functions_launcher/src/lookup.rs (hunk 1 above)
Comment (mariaschett) [reply oak_functions_launcher/src/lookup.rs:45]: Clarified in a comment, but this was meant to be a small PR just to compute the chunks, but not to send them yet.

**Review by mariaschett — COMMENTED** [review]
Thank you!

**Pull request merged**